src/store.c

```diff
@@ -79,29 +88,80 @@ struct rill_store
 // -----------------------------------------------------------------------------
 
 static struct encoder store_encoder(
-        struct rill_store *store, struct indexer *indexer)
+        struct rill_store *store,
+        struct indexer *indexer,
+        struct vals* vals,
+        uint64_t offset)
 {
     return make_encoder(
-            store->vma + store->head->data_off,
+            store->vma + offset,
             store->vma + store->vma_len,
-            store->vals, indexer);
+            vals,
+            indexer);
 }
 
-static struct decoder store_decoder(struct rill_store *store)
+static struct decoder store_decoder(
```

Comment:
This function should probably just end up calling `store_decoder_at` as there's a lot of redundancy between the two functions.

Reply:
Will do this as soon as I get the other code path that uses `_at` to work properly!